specs/WebAuthenticationRequested.md

```diff
@@ -0,0 +1,302 @@
+{
+    args.Cancel = true;
+};
+```
+
+## Read authorization challenge string
+Developer can read the authorization challenge string sent by server. Note that if the developer doesn't cancel or provide a response, the login dialog prompt will be shown to the user.
+
+```cpp
+webviewStaging2->add_WebAuthenticationRequested(
+    Callback<ICoreWebView2StagingWebAuthenticationRequestedEventHandler>(
+        [this](
+            ICoreWebView2Staging2* sender,
+            ICoreWebView2StagingWebAuthenticationRequestedEventArgs* args) {
+            args->get_Challenge(&challenge);
+            if (wcsncmp(challenge.get(), L"Expected login credentials") != 0) {
+                args->put_Cancel(true);
+            } 
+            return S_OK;
+        })
+        .Get(),
+    &m_webAuthenticationRequestedToken));
+```
+
+```c#
+webView.CoreWebView2.WebAuthenticationRequested += delegate (object sender, CoreWebView2WebAuthenticationRequestedEventArgs args)
+{
+    if (args.Challenge.Equals("Expected login credentials")) {
+        args.Cancel = true;
+    } 
+};
+```
+
+# Remarks
+
+
+# API Notes
+
+
+# API Details
+```idl
+/// This is the ICoreWebView2 Staging interface.
+[uuid(9EAFB7D0-88C3-4450-BBFB-C05A46C40C72), object, pointer_default(unique)]
+interface ICoreWebView2Staging2 : IUnknown {
+  /// Add an event handler for the WebAuthenticationRequested event.
+  /// WebAuthenticationRequested event fires when WebView encountered a Basic HTTP
+  /// Authentication request.
+  ///
+  /// The host can provide a response with credentials for the authentication or
+  /// cancel the request. If the host doesn't handle the event, WebView will show
+  /// the authorization challenge dialog prompt to user.
+  ///
+  HRESULT add_WebAuthenticationRequested(
+    [in] ICoreWebView2StagingWebAuthenticationRequestedEventHandler* eventHandler,
+    [out] EventRegistrationToken* token);
+  /// Remove an event handler previously added with add_WebResourceRequested.
+  HRESULT remove_WebAuthenticationRequested(
+      [in] EventRegistrationToken token);
+}
+
+/// This is the CoreWebView2WebAuthenticationRequestedEventHandler Staging interface
+[uuid(f87e5d35-3248-406b-81dd-1c36aab8081d), object, pointer_default(unique)]
+interface ICoreWebView2StagingWebAuthenticationRequestedEventHandler : IUnknown
+{
+  /// Called to provide the implementer with the event args for the
+  /// corresponding event.
+  HRESULT Invoke(
+      [in] ICoreWebView2Staging2* sender,
+      [in] ICoreWebView2StagingWebAuthenticationRequestedEventArgs* args);
+}
+
+/// Represents a Basic HTTP authentication response that contains a user name
+/// and a password as according to RFC7617 (https://tools.ietf.org/html/rfc7617)
+[uuid(bc9cfd60-29c4-4943-a83b-d0d2f3e7df03), object, pointer_default(unique)]
+interface ICoreWebView2StagingWebAuthenticationResponse : IUnknown
+{
+  /// User name provided for authorization.
+  [propget] HRESULT UserName([out, retval] LPWSTR* userName);
```

Comment:
Are there any restrictions on what's allowed in the username or password fields? If so, what happens if the end dev provides invalid values?

Reply:
I had referenced the RFC link for the specification of the username password above. Do we need to specify any behavior for the case the developer doesn't go by specification?